frontend/app/src/components/asset-manager/RestoreAssetDbButton.vue

```diff
@@ -1,15 +1,84 @@
-      {{ t('asset_update.restore.action') }}
-    </v-btn>
+    <template v-if="dropdown">
+      <v-menu offset-y>
+        <template #activator="{ on }">
+          <v-btn color="primary" depressed v-on="on">
+            {{ t('asset_update.restore.title') }}
+          </v-btn>
+        </template>
+        <v-list>
+          <v-list-item two-line link @click="restoreClick('soft')">
+            <v-list-item-content>
+              <v-list-item-title>
+                {{ t('asset_update.restore.soft_reset') }}
+              </v-list-item-title>
+              <v-list-item-subtitle>
+                {{ t('asset_update.restore.soft_reset_hint') }}
+              </v-list-item-subtitle>
+            </v-list-item-content>
+          </v-list-item>
+          <v-list-item two-line link @click="restoreClick('hard')">
+            <v-list-item-content>
+              <v-list-item-title>
+                {{ t('asset_update.restore.hard_reset') }}
+              </v-list-item-title>
+              <v-list-item-subtitle>
+                {{ t('asset_update.restore.hard_reset_hint') }}
+              </v-list-item-subtitle>
+            </v-list-item-content>
+          </v-list-item>
+        </v-list>
+      </v-menu>
+    </template>
+    <template v-else>
+      <v-tooltip top max-width="200">
+        <template #activator="{ on }">
+          <v-btn
+            outlined
+            depressed
+            color="primary"
+            v-on="on"
+            @click="restoreClick('soft')"
+          >
+            {{ t('asset_update.restore.soft_reset') }}
+          </v-btn>
+        </template>
+        <span>{{ t('asset_update.restore.soft_reset_hint') }}</span>
+      </v-tooltip>
+      <v-tooltip top max-width="200">
+        <template #activator="{ on }">
+          <v-btn
+            class="ml-4"
+            depressed
+            color="primary"
+            v-on="on"
+            @click="restoreClick('hard')"
+          >
+            {{ t('asset_update.restore.hard_reset') }}
+          </v-btn>
+        </template>
+        <span>{{ t('asset_update.restore.hard_reset_hint') }}</span>
+      </v-tooltip>
+    </template>
     <confirm-dialog
       :display="confirmRestore"
       :title="tc('asset_update.restore.delete_confirmation.title')"
-      :message="tc('asset_update.restore.delete_confirmation.message')"
+      :message="
+        resetType === 'soft'
+          ? tc('asset_update.restore.delete_confirmation.soft_reset_message')
+          : tc('asset_update.restore.delete_confirmation.hard_reset_message')
+      "
       @confirm="restoreAssets"
       @cancel="confirmRestore = false"
     />
+    <confirm-dialog
+      :display="doubleConfirmation"
+      :title="tc('asset_update.restore.hard_restore_confirmation.title')"
+      :message="tc('asset_update.restore.hard_restore_confirmation.message')"
+      @confirm="confirmReset"
+      @cancel="doubleConfirmation = false"
+    />
```

Comment:
@lukicenturi I have had it in my mind for a while now, but we should start thinking of a way to have a central confirmation dialog instead of adding it to every component that needs to confirm something. 

I will try something at some point but if you also have any ideas on that direction please let me know.